ports/stm32/boards/common_n6_flash.ld

```diff
@@ -0,0 +1,83 @@
+/* Memory layout for N6 when the application runs from external flash in XIP mode.
+
+    FLASH_APP   .isr_vector
+    FLASH_APP   .text
+    FLASH_APP   .data
+
+    RAM         .data
+    RAM         .bss
+    RAM         .heap
+    RAM         .stack
+*/
+
+ENTRY(Reset_Handler)
+
+REGION_ALIAS("FLASH_ISR", FLASH_APP);
+REGION_ALIAS("FLASH_COMMON", FLASH_APP);
+
+/* define output sections */
+SECTIONS
+{
+    .isr_vector :
+    {
+        . = ALIGN(4);
+        __isr_vector_ram_start = .;
+
+        KEEP(*(.isr_vector)) /* Startup code */
+
+        /* These functions need to run from ram to enable uart
```

Comment:
I assume this is because IRQs are being disabled for too long? If that’s the case, would it help to re-enable IRQs after each block erase/write operation, rather than keeping them disabled across all blocks?  If I’m not wrong, this is how it's done in the mimxrt and rp2 ports, and it might help avoid the need to move all those functions and their data to RAM.

Reply:
Actually, the UART functions don't need to go in RAM anymore.  I've fully disabled IRQs during XSPI flash erase/write.  It didn't seem to work if I just raised the IRQ priority level.

And the code in `spibdev.c` does enable IRQs after each block erase.